tests/lax_test.py

```diff
@@ -1748,6 +1748,37 @@ def testReduce(self, op, init_val, shape, dtype, dims):
     args_maker = lambda: [rng(shape, dtype)]
     self._CompileAndCheck(fun, args_maker)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_op={}_shape={}_reducedims={}_initval={}_prim={}"
+       .format(op.__name__, shape, dims, init_val, prim),
+       "op": op, "init_val": init_val, "shape": shape, "dims": dims, "prim": prim}
+      for init_val, op , prim in [
+        (True, lax.bitwise_and, jax.lax.reduce_and_p),
+        (False, lax.bitwise_or, jax.lax.reduce_or_p),
+        (False, lax.bitwise_xor, jax.lax.reduce_xor_p),
+      ]
+      for shape, dims in [
+          [(3, 4, 5), (0,)], [(3, 4, 5), (1, 2)],
+          [(3, 4, 5), (0, 2)], [(3, 4, 5), (0, 1, 2)]
+      ]))
+  def testReduceBoolean(self, op, init_val, shape, dims, prim):
+    dtype = np.bool_
+    rng = jtu.rand_bool(self.rng())
+    init_val = np.asarray(init_val, dtype=dtype)
+    fun = lambda operand, init_val: lax.reduce(operand, init_val, op, dims)
+    args_maker = lambda: [rng(shape, dtype), init_val]
+    self._CompileAndCheck(fun, args_maker)
```

Comment:
Let's also do a `self._CheckAgainstNumpy` here and below for a suitable numpy version – we want to make sure the functions are returning the correct result!

Reply:
Done! I modeled it after the ReduceWindow test below.